zkevm-circuits/src/evm_circuit/execution/end_tx.rs

```diff
@@ -70,6 +72,15 @@ impl<F: Field> ExecutionGadget<F> for EndTxGadget<F> {
         cb.tx_refund_read(tx_id.expr(), refund.expr());
         let effective_refund = MinMaxGadget::construct(cb, max_refund.quotient(), refund.expr());
 
+        // gas_used & refund == 0 if tx is invalid
+        cb.condition(is_tx_invalid.expr(), |cb| {
+            cb.require_zero("gas_used == 0 if tx is invalid", gas_used.expr());
```

Comment:
The spec does not contain this check and I believe it is not required. If you think so to better to remove it so we don't diverge from the spec.